### PR TITLE
Add connection with backend.

### PR DESCRIPTION
**Pull request by martinalturria**
- Add all necessary files using clean architecture (Domain, infrastructure, Providers)
- Add entities and user repositories to the domain
- Add in Infrastructure the implementation of the repository and the mappers corresponding to users.
- Add the Provider folder and the providers corresponding to users.
- Modify Users view to show them from the Backend

### CODE REVIEW

File: lib/domain/repositories/user_repository.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import '../entities/user_entity.dart';
+
+abstract class UserRepository {
+  Future<List<UserEntity>> getUsers();
+  Future<UserEntity> getUserById(String id);
+  Future<UpdateResult> updateUser(UserEntity user);
+  Future<UpdateResult> updateStatus(String email, String status);
+  // Otros métodos del repositorio
```
Comment (manduinca) [inline lib/domain/repositories/user_repository.dart]: This comment is not necessary.

File: lib/Screens/UserScreen/Client/client_screen.dart [hunk 2]
```diff
@@ -20,70 +22,54 @@ class ClientScreen extends ConsumerWidget {
...
+                        vertical: 15, horizontal: 25),
+                    child: ListTile(
+                      onTap: () {
+                        // Acción al hacer clic en un usuario
+                      },
+                      onLongPress: () {
+                        showAlert(context, ref);
+                      },
+                      title: Text(
+                          "Nombre: ${user.name} - Empresa: ACA IRIA LA EMPRESA CON LA RELACION"),
```
Comment (manduinca) [inline lib/Screens/UserScreen/Client/client_screen.dart:47]: will it showed to users?

File: lib/Screens/UserScreen/Client/client_create_screen.dart [hunk 3]
```diff
@@ -211,41 +212,31 @@ Widget buttons(BuildContext context, ref, emailController, nameController, enter
...
     barrierDismissible: false,
-    context: context, 
-    builder: (BuildContext context) { 
+    context: context,
+    builder: (BuildContext context) {
       return AlertDialog(
         title: const Text('Registrar'),
-        content: const Text('¿Estas Seguro Que Deseas Registrar Este Cliente?', style: TextStyle(fontSize: 17),),
+        content: const Text(
+          '¿Estas Seguro Que Deseas Registrar Este Cliente?',
```
Comment (manduinca) [inline lib/Screens/UserScreen/Client/client_create_screen.dart]: Better: ¿Estás seguro que deseas registrar este usuario?

File: lib/Screens/UserScreen/Client/client_screen.dart (hunk 2 above)
Comment (martinalturria) [reply lib/Screens/UserScreen/Client/client_screen.dart:47]: yes, that's how they had done the view, the related companies would be shown to the users

File: lib/domain/repositories/user_repository.dart (hunk 1 above)
Comment (martinalturria) [reply lib/domain/repositories/user_repository.dart]: Fixed, I had left it to remember the user post method that we talked about this morning.

**Comment by martinalturria** [comment]
Fix the requested

**Review by manduinca — APPROVED** [review]

**Review by brunofoster03 — APPROVED** [review]
Ok:)

**Pull request merged**